lib/galaxy/tools/data/__init__.py

```diff
@@ -517,7 +526,8 @@ def parse_file_fields(self, reader, errors=None, here="__HERE__"):
         TODO: Allow named access to fields using the column names.
         """
         separator_char = "<TAB>" if self.separator == "\t" else self.separator
-
+        if isinstance(reader, str):
```

Comment:
I completely agree with **not** doing `open(filename`) on an argument in a method call (your edit on line 518). However, now that it's a filename as it should be, I think the argument name should be changed accordingly. Also, I suggest treating it as a string (i.e., not doing "if the reader is a string, make it into a reader, otherwise assume it's a reader"). If there's other code that calls this method passing a file handle instead, my suggestion is to change that other code, or refactor this file to make the 2 options (filename vs. file handle) more explicit (maybe separate keyword args, etc...)

Reply:
Definitely agree that passing a string to a variable named `reader` is not super great, definitely worth a polish. Thanks for these thorough reviews @ic4f !